pkg/generate.go

```diff
@@ -56,15 +69,45 @@ func Generate(fs afero.Afero, gitOpsFolder string, outputFolder string, componen
 		deploymentFileName: deployment,
 	}
 
+	var service *corev1.Service
+	var route *routev1.Route
 	// If a targetPort was specified, also generate a service and route
-	if component.TargetPort != 0 {
-		service := generateService(component)
-		route := generateRoute(component)
+	if len(component.KubernetesResources.Services) == 0 && len(component.KubernetesResources.Routes) == 0 && component.TargetPort != 0 {
```

Comment:
if `spec.targetPort` isn't present in any of the service resources in the outerloop yaml(s), do we still need to generate a service?

Reply:
so this `component.TargetPort` in this line here is from the `Component` Spec CR. Its not the targetPort in any of the outerloop Kubernetes YAMLs. The outerloop Kubernetes YAML values are in `component.KubernetesResources.Services` and `component.KubernetesResources.Routes`

So with this if statement, we generate service and route if both service AND route definitions are not present in the outerloop YAML file AND if a TargetPort has been defined in the `Component.Spec` CR (which is what we do in the main branch, if Component CR Spec TargetPort is not 0, generate Service and Route)